packages/docs/docs/entities/components-repo.md

```diff
@@ -1,30 +1,90 @@
-# ComponentsRepo
+# The Components Repository
 
-`ComponentsRepo` is class which accepts string `name` and optional `getType` function as constructor parameter.
+The component repository is a map of data type to AutoView React components used to render a field of that data type.
 
-`ComponentsRepo` instance should be provided to the `AutoView` context.
+In essence, the component repository is a map of
+```
+Map<type => React.ComponentType<AutoViewProps>>
+```
 
-```js
-import {
-  RepositoryProvider,
-  AutoView
-} from '@autoviews/core';
+An AutoView React Component has the signature
+```
+React.ComponentType<AutoViewProps>
+```
 
-//...
-return (
-    <RepositoryProvider components={myRepo}>
-        <AutoView {...props} />
-    </RepositoryProvider>
-)
+`AutoViewProps` has a lot of properties, but the most important is the `data` prop which is the field data the component has to render.
+     
+## The ComponentsRepo Class
+                             
+The `ComponentsRepo` is the main implementation of the Component Repository.
+It takes two parameters
+* `name`: Repository name
+* `getNodeType`: the callback that resolves the `type` from the `JSONSchema` leaf (more on the callback below)
+
+### Example - A simple repository example
+
+```typescript
+import { ComponentsRepo } from "@autoviews/core";
+
+export const myRepo = new ComponentsRepo("displayRepo")
+  .register("string", {
+    name: "textComponent",
+    component: (props) => <span>{props.data}</span>
+  })
+  .register("number", {
+    name: "numberComponent",
+    component: (props) => <span>{props.data}</span>
+  })
+  .register("boolean", {
+    name: "booleanComponent",
+    component: (props) => <span>{props.data ? "+" : "-"}</span>
+  });
+```
+
+## The register function
+The repository register function adds a new component to the repository per data type.
+```typescript
+public register(type: string | symbol, 
+  record: ComponentRepoRecord<AutoViewProps>)
+
+export interface ComponentRepoRecord<P> {
+    name: string;
+    component: React.ComponentType<P>;
+    predicate?: Predicate;
+    defaultOptions?: any;
+}
+```
+
+Where
+* `type`: the name of the `JSONSchema`'s type, such as `string`, `object`, `number`
```

Comment:
`register` method also takes symbols as type. That's for components, which shouldn't be resolved with `getNodeType`, but they're also stored in repository and could be obtained via context. I.e. some components which are not independent, like implementation of `oneOf`, which should be rendered only inside others component's implementation.

Reply:
added